sunpy/instr/aia/response.py

```diff
@@ -0,0 +1,308 @@
+
+        Parameters
+        ----------
+        include_crosstalk : `bool`
+            See `calculate_crosstalk`
+        """
+        Response = namedtuple('Response', 'wavelength response')
+        wavelength_response = {}
+        for channel in self._channel_info:
+            system_gain = self.calculate_system_gain(channel)
+            effective_area = self.calculate_effective_area(channel,
+                                                           include_crosstalk=include_crosstalk)
+            wavelength_response[channel] = Response(
+                wavelength=self._channel_info[channel]['wavelength'],
+                response=system_gain*effective_area)
+
+        return wavelength_response
+
+    def calculate_temperature_response(self, ion_list=None, emiss_table_file=None, **kwargs):
+        """
+        Calculate the temperature response functions.
+
+        According to [1]_, the temperature response function for channel :math:`i` is
+        given by,
+
+        .. math::
+           K_i(T) = \int_0^{\infty}\mathrm{d}\lambda\,G(\lambda,n,T)R_i(\lambda),
+
+        in units of DN :math:`\mathrm{cm}^{5}\,\mathrm{s}^{-1}\,\mathrm{pixel}^{-1}` where
+        :math:`R_i(\lambda)` is the wavelength response function of channel :math:`i` and
+        :math:`G(\lambda,T)` is the contribution function. Note that :math:`G` includes both
+        the line and continuum contributions such that,
+
+        .. math::
+           G(\lambda,n,T) = G_{ff}(\lambda,T) + G_{fb}(\lambda,T) + G_{tp}(\lambda,T) + \sum_{Z,z}\\frac{1}{4\pi}\mathrm{Ab}(Z)\\frac{N(Z^{+z})}{N(Z)}\\varepsilon_{Z,z}(\lambda,n,T)\\frac{1}{n}
+
+        where :math:`G_{ff}(\lambda,T),\,G_{fb}(\lambda,T),\,G_{tp}(\lambda,T)` are the free-free,
+        free-bound, and two-photon continuum emissivities.
+
+        Parameters
+        ----------
+        ion_list : array-like, optional
+            List of ion names, e.g. 'fe_15' for Fe XV. Defaults to all ions in CHIANTI database
+        emiss_table_file : `str`, optional
+            Defaults to 'aia_emiss_table.h5' in the default download directory for SunPy
+
+
+        .. warning::
+            The first time you calculate the temperature response functions, you will
+            need to build the emission table by calculating the contribution function for each
+            ion. This may take up to 30 minutes, but will only need to be done once.
+        """
+        if emiss_table_file is None:
+            emiss_table_file = os.path.join(get_and_create_download_dir(), 'aia_emiss_table.h5')
+        if not os.path.exists(emiss_table_file):
+            warnings.warn('Building emissivity table {}. This may take a few minutes, but only needs to be done once.'.format(emiss_table_file))
+            make_emiss_table(emiss_table_file, MasterList)
+
+        wavelength_response = self.calculate_wavelength_response(include_crosstalk=kwargs.get('include_crosstalk', True))
```

Comment:
I did this in the effective_area function too. I think I was originally trying to avoid repeating the `include_crosstalk` kwarg explicitly since it gets immediately passed to another function, but this obscures the fact that include_crosstalk can be passed at all.